stl/inc/span

```diff
@@ -0,0 +1,723 @@
+
+    // [span.sub] Subviews
+    template <size_t _Count>
+    _NODISCARD constexpr span<element_type, _Count> first() const noexcept {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Count <= _Extent, "Count out of range in span::first()");
+        }
+#if _CONTAINER_DEBUG_LEVEL > 0
+        else {
+            _STL_VERIFY(_Count <= size(), "Count out of range in span::first()");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {data(), _Count};
+    }
+
+    _NODISCARD constexpr span<element_type, dynamic_extent> first(const index_type _Count) const noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Count <= size(), "Count out of range in span::first(count)");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {data(), _Count};
+    }
+
+    template <size_t _Count>
+    _NODISCARD constexpr span<element_type, _Count> last() const noexcept {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Count <= _Extent, "Count out of range in span::last()");
+        }
+#if _CONTAINER_DEBUG_LEVEL > 0
+        else {
+            _STL_VERIFY(_Count <= size(), "Count out of range in span::last()");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {data() + size() - _Count, _Count};
+    }
+
+    _NODISCARD constexpr span<element_type, dynamic_extent> last(const index_type _Count) const noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Count <= size(), "Count out of range in span::last(count)");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {data() + size() - _Count, _Count};
+    }
+
+    template <size_t _Offset, size_t _Count = dynamic_extent>
+    _NODISCARD constexpr span<element_type,
+        _Count != dynamic_extent ? _Count : (_Extent != dynamic_extent ? _Extent - _Offset : dynamic_extent)>
+        subspan() const noexcept {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Offset <= _Extent, "Offset out of range in span::subspan()");
+            static_assert(
+                _Count == dynamic_extent || _Offset <= _Extent - _Count, "Count out of range in span::subspan()");
+        }
+#if _CONTAINER_DEBUG_LEVEL > 0
+        else {
+            _STL_VERIFY(_Offset <= size(), "Offset out of range in span::subspan()");
+            _STL_VERIFY(
+                _Count == dynamic_extent || _Offset <= size() - _Count, "Count out of range in span::subspan()");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return {data() + _Offset, _Count == dynamic_extent ? size() - _Offset : _Count};
+    }
+
+    _NODISCARD constexpr span<element_type, dynamic_extent> subspan(
+        const index_type _Offset, const index_type _Count = dynamic_extent) const noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Offset <= size(), "Offset out of range in span::subspan(offset, count)");
+        _STL_VERIFY(_Count == dynamic_extent || _Offset <= size() - _Count,

```

Comment:
```suggestion
        _STL_VERIFY(_Count == dynamic_extent || _Count <= size() - _Offset,
```

Reply:
Yepp

Reply:
This change is still necessary.

Reply:
must have been lost when commiting. Fixed for good